### DISCUSSION TITLE
[New Feature] Subterranean defenses

### DISCUSSION THREAD

**Discussion by CCTiberius** [Vinifera-Developers/Vinifera#1499] [Feature Requests]
### Description:
Defenses that go down into the ground when not firing, exactly like the TD SAMs or how the Nod laser turret in TS was supposed to work.

### Possible Implementation:
When a defense is below ground, it should have a separate armour type, so modders can choose to make them harder to destroy when burried. They should not go up and down between shots, but ideally stay up as long as they are in combat (maybe coded so as long as they have a target they don't lower), to prevent it inadvertently lowering a defenses RoF. If powered down a defense with this logic should be burried, so a player can make a tactical choice of powering off a dying defense to save it, making gameplay more dynamic.

**Comment by Flame-Liekki** [comment]
That would be nice.  Especially if it wasn't specifically underground and above ground, but just allowing turrets to have idle animations/transition animations/combat animations.  Possibly could allow for TD-like obelisks as-well.

**Comment by Rampastring** [comment]
It is noteworthy that while this logic exists in TD, it is quite buggy there, as the Nod SAM sites pop in and back out between every missile.

**Comment by CCHyper** [comment]
> It is noteworthy that while this logic exists in TD, it is quite buggy there, as the Nod SAM sites pop in and back out between every missile.

Sounds like a waiting time before it stops scanning _(I guess visually scanning for targets)_ would solve this.

**Comment by CCTiberius** [comment]
@Flame-Liekki I'd think those are separate logics, but yes, both would be awesome to have. I don't see anything preventing modders from using this logic in a wider application though, it's just a matter of what model and animation you give the defense. For example you could have a defense's "underground/non-combat" state be a protective shield, or it simply looking like it's "off" if it's an energy weapon. If you don't want changes to its sturdiness, just define the non-combat/"dug-in" armour the same as the in-combat one.
